dotnet/src/Connectors/Connectors.AI.OpenAI/TextToImage/AzureOpenAITextToImageService.cs

```diff
@@ -3,209 +3,173 @@
+    private readonly OpenAIClient _client;
+    private readonly ILogger _logger;
 
     /// <summary>
-    /// Generation Image Operation path
-    /// </summary>
-    private const string GenerationImageOperation = "openai/images/generations:submit";
-
-    /// <summary>
-    /// Get Image Operation path
-    /// </summary>
-    private const string GetImageOperation = "openai/operations/images";
-
-    /// <summary>
-    /// Azure OpenAI REST API endpoint
-    /// </summary>
-    private readonly string _endpoint;
-
-    /// <summary>
-    /// Azure OpenAI API key
-    /// </summary>
-    private readonly string _apiKey;
-
-    /// <summary>
-    /// Maximum number of attempts to retrieve the text to image operation result.
-    /// </summary>
-    private readonly int _maxRetryCount;
-
-    /// <summary>
-    /// Azure OpenAI Endpoint ApiVersion
-    /// </summary>
-    private readonly string _apiVersion;
-
-    /// <summary>
-    /// Create a new instance of Azure OpenAI text to image service
+    /// Create a new instance of Azure OpenAI image generation service
     /// </summary>
     /// <param name="endpoint">Azure OpenAI deployment URL, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
-    /// <param name="modelId">Azure OpenAI model id, see https://learn.microsoft.com/azure/cognitive-services/openai/how-to/create-resource</param>
+    /// <param name="modelId">Model identifier, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
     /// <param name="apiKey">Azure OpenAI API key, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
     /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
     /// <param name="loggerFactory">The ILoggerFactory used to create a logger for logging. If null, no logging will be performed.</param>
-    /// <param name="maxRetryCount"> Maximum number of attempts to retrieve the text to image operation result.</param>
+    /// <param name="maxRetryCount"> Maximum number of attempts to retrieve the image generation operation result.</param>
     /// <param name="apiVersion">Azure OpenAI Endpoint ApiVersion</param>
     public AzureOpenAITextToImageService(
-        string? endpoint, string modelId, string apiKey, HttpClient? httpClient = null, ILoggerFactory? loggerFactory = null, int? maxRetryCount = null, string? apiVersion = null)
+        string? endpoint,
+        string modelId,
+        string apiKey,
+        HttpClient? httpClient = null,
+        ILoggerFactory? loggerFactory = null,
+        int? maxRetryCount = null,
+        string? apiVersion = null)
     {
         Verify.NotNullOrWhiteSpace(apiKey);
-        Verify.NotNull(modelId);
+
+        if (!string.IsNullOrEmpty(modelId))
+        {
+            throw new NotSupportedException($"Specifying a model ID, such as Dall-E-3, is not supported yet. Please provide an empty '{nameof(modelId)}' to use Dall-E-2.");
```

Comment:
If DallE2 is the only model, which is supported by this class, I think we should remove `modelId` parameter from constructor, because with this implementation parameter `modelId` should be always set as empty, which would be nice to avoid. Since DallE3 uses a different HTTP flow compared to Dalle2, maybe this logic will be implemented in separate class, so `modelId` wouldn't be required in the future for this class.

Another option would be to set default value for `modelId` parameter as `dall-e-2` and in case of other Model ID - temporarily throw `NotSupportedException` exception until we add required implementation for DallE3. 

The main point here is to avoid accepting empty value in required constructor parameter as the only valid option.

Reply:
In this case, as I mentioned, we can set `modelId` default value to `dall-e-2` and throw exception if passed ID is different. We don't know when support for DallE3 will be added, so there is a chance that empty parameter will be required to pass for a long period of time.

Reply:
"dall-e-2" exists only on OpenAI though, not on Azure where the model ID is undefined. Under Azure Studio Deployments, dall-e-2 doesn't show, it's more like a special endpoint. With dall-e-3 instead the models appears in the list.

Thoughts?

Reply:
screenshot for DallE3

![image](https://github.com/microsoft/semantic-kernel/assets/371009/e4fc6273-4ad7-45e4-9371-388f375caa6e)

Reply:
In the future when we add support for DallE3 in the same class, how users will differentiate between DallE2 and DallE3? If DallE3 will require `dall-e-3` model id, then intuitively user will try to set `dall-e-2` model id for DallE2. Even if `dall-e-2` is not part of Azure Studio Deployments, we still have to make it easy for SK users. 

Another way, as mentioned in previous comment, would be to implement DallE3 in scope of separate class, especially if this model has different HTTP flow than DallE2. In this case, for DallE2 implementation we can remove `modelId` parameter completely. We can also rename the class from `AzureOpenAITextToImageService` to `AzureOpenAIDalle2Service` or something like that, to clearly specify that this class works with special endpoint for DallE2. cc: @markwallace-microsoft 

But I still think that empty parameter is not clear identification that this is for DallE2.

Reply:
change the code, now it let's everything through except for "dall-e-3"

Reply:
Thank you!